src/plots/linePlot.ts

```diff
@@ -425,6 +450,74 @@ export module Plots {
       return filteredDataIndices;
     }
 
+    private _filterDownsampling(dataset: Dataset, indices: number[]) {
+      let xAccessor = this.x().accessor;
+      let yAccessor = this.y().accessor;
+      let filteredIndices = this._filterDownsamplingSlope(dataset, indices, this.x().scale, this.y().scale, xAccessor, yAccessor);
+      return filteredIndices;
+    }
+
+    private _filterDownsamplingSlope(dataset: Dataset, indices: number[],
+      xScale: Scale<any, number>, yScale: Scale<any, number>, xAcessor: Accessor<any>, yAcessor: Accessor<any>) {
+      let data = dataset.data();
+      let filteredIndices: number[] = [];
+      let min: number;
+      let max: number;
+      let currentSlope: number;
+      if (indices.length === 0) {
+        return filteredIndices;
+      }
+      filteredIndices.push(indices[0]);
+      function belongToCurBucket(i: number) {
+        let p1x = xScale.scale(xAcessor(data[indices[i]], indices[i], dataset));
+        let p1y = yScale.scale(yAcessor(data[indices[i]], indices[i], dataset));
+        let p2x = xScale.scale(xAcessor(data[indices[i + 1]], indices[i + 1], dataset));
+        let p2y = yScale.scale(yAcessor(data[indices[i + 1]], indices[i + 1], dataset));
+        if (currentSlope == null) {
+          currentSlope = (Math.floor(p1x) === Math.floor(p2x)) ? Infinity : (p2y - p1y) / (p2x - p1x);
+          min = (currentSlope === Infinity) ? yAcessor(data[indices[i]], indices[i], dataset) :
```

Comment:
`p1y` is the scaled y value, which means the maximum of `p1y` actually has the smallest y value in dataset. But as scale is a linear function, this shouldn't be a problem. I change  `yAcessor(data[indices[i]], indices[i], dataset)` to `p1y`.